lib/heartcheck/app.rb

```diff
@@ -16,6 +16,7 @@ class App
       '/functional' => Controllers::Functional,
       '/dev' => Controllers::Dev,
       '/info' => Controllers::Info,
+      '/inspect' => Controllers::Inspect,
```

Comment:
This new route will provide information without activate the check's and dont change the response for compatibility. However, we are planning modify to include that kind of information to improve more.

The main feature is like as described in begin, a possibility to create a big map of dependencies of all application who are connected.

Reply:
Could you give us an example of the output of this new endpoint? I think it will be easier to understand the purpose of this.

Reply:
Example output: 

```
{
  "application_name": "CloudPrivateApi::Application",
  "environment": "development",
  "dependencies": [
    {
      host: 'someservice.sysint.locaweb.com.br',
      port: 80,
      schema: http
    },
    {
      "error": "Heartcheck::Checks::Sidekiq #url_info not implemented."
    },
    ...
  ]
}

So the idea is not to run or get check status, but to provide a 'skeletton' of the current aplication, based on the heartcheck checks. Using something like that, we can crawl through our infrastructure and create a map (visual or not) of our applications and their dependencies. This is something not currently attainable by the existent endpoints, because they group all Firewall tests in a single output and don't give enough detail about all tests.

This endpoint and the '#uri_info' is a non-intrusive and non-api-breaking (?) way to get this information from all kinds of checks without significant changes to all projects, besides bumping up the heartcheck version.

Reply:
Yes.. this new route is a better way to format all endpoints used by all checks. The clients does not need to configure anything in your config file, just update the gem and the new route was be available with that information.

Reply:
> About this point, this is not required, it is possible (and I think better) to have one firewall check for each service, like this:

The problem is that Heartcheck allows it and many projects do it:
```ruby
monitor.add :firewall do |c|
  c.add_service(url: CloudCoreConfig[:cas_system2][:sys_url]) # Cas System
  c.add_service(url: CloudCoreConfig[:cas_system2][:user_url]) # Cas Web
  c.add_service(url: CloudCoreConfig[:server_registry][:base_url]) # Cegonha
  ...
end
```
So the realistic workaround is just deal with it and break the firewall tests into separate dependencies, IMHO. It just works for every case and causes no super ugly implementation.

> Won't be necessary to add new entries specifically for the inspect endpoint, right? (when I had talked about duplication, was considering I would add new entries)
> 
> And won't be necessary to change existing checks in the heartcheck initializer of the apps because each plugin will know how to return the "reference" (URL, IP, etc) of your dependency(ies), right?

Exactly. This is the major driver of the approach we took. We will just need that all applications bump up their Heartcheck versions and set a env var, in their own time, without breaking anything.

Reply:
great! :+1:

Reply:
> we can crawl through our infrastructure and create a map (visual or not) of our applications and their dependencies

based on what you said, what about changing the route `/inspect` to `/dependencies_map` (or something like this)?

Reply:
Another point is you are basing your feature in `:firewall` check, but some apps don't use this check and prefer to use the [heartcheck-webservice](https://github.com/locaweb/heartcheck-webservice) check. 

Are planning to implement something similar in this gem? (Otherwise, it will not show these dependencies)

Reply:
> based on what you said, what about changing the route /inspect to /dependencies_map (or something like this)?

Well, it has little more than the dependencies, it bears the application name, used to have the environment variable (🤣 ) and it was devised to include more information needed in the future, hence we kept the "inspect" term.

> Are planning to implement something similar in this gem? (Otherwise, it will not show these dependencies)

Yes, there are several implementations already (webservices, redis, activerecord...) just waiting for this PR to be PRed too. There will probably be missing implementations (checks we don't use in our projects), but those should be implemented when needed, too. There are some words about that in the commit and PR description:

> All check plug-ins must implement their own version of #uri_info
to be shown in the dependencies collection. Such implementations
will be provided soon in future PRs.

Reply:
@luizcavalcanti ok great! For me it is ok